env0/resource_environment.go

```diff
@@ -23,27 +23,142 @@ func resourceEnvironment() *schema.Resource {
 			},
 			"name": {
 				Type:        schema.TypeString,
-				Description: "The environment's name",
+				Description: "the environment's name",
 				Required:    true,
 			},
 			"project_id": {
 				Type:        schema.TypeString,
 				Description: "project id of the environment",
 				Required:    true,
+				ForceNew:    true,
 			},
 			"template_id": {
 				Type:        schema.TypeString,
 				Description: "the template id the environment is to be created from",
 				Required:    true,
 			},
+			"workspace": {
+				Type:        schema.TypeString,
+				Description: "the terraform workspace of the environment",
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"revision": {
+				Type:        schema.TypeString,
+				Description: "the revision the environment is to be run against",
+				Optional:    true,
+				Computed:    true,
+			},
```

Comment:
Please notice that these two are computed
I wanted those fields to override those values when specified.
Without computed, when they were left empty after a read would happen ( happens on plan\apply\refresh etc)
they were set to the terraform state (since the api returned values for them).
So, without actually changing the terraform config it would appear as if you changed them back to null.

Basically this is the only method I found to handle anything that may be overridden (and more importantly can be left empty) and is generated on the backend if it's not on the request.

*This may cause problems with drift detection

An example of what happened without computed: 

terraform apply this tf config
```
resource "env0_environment" "test" {
  name        = "env name"
  project_id  = "ed7c54d2-ce49-4ad9-ab4e-be5f62107f03"
  template_id = "6a030803-4643-4ad0-a1fe-037feed14f21"
}
```
terraform plan (without changing anything)
terraform would perform a read 
setting the resource to
```
resource "env0_environment" "test" {
  name        = "env name"
  project_id  = "ed7c54d2-ce49-4ad9-ab4e-be5f62107f03"
  template_id = "6a030803-4643-4ad0-a1fe-037feed14f21"
  workspace = "env0ae77fe"
}
```
then it would show that the plan would be
```
resource "env0_environment" "test" {
  name        = "env name"
  project_id  = "ed7c54d2-ce49-4ad9-ab4e-be5f62107f03"
  template_id = "6a030803-4643-4ad0-a1fe-037feed14f21"
  workspace = "env0ae77fe" -> null
}
```
because no workspace was specified in the tf config.

Reply:
Thanks for the explanation 👑